raw.go

```diff
@@ -31,3 +31,32 @@ type RawSpan struct {
 	// The span's associated baggage.
 	Baggage map[string]string // initialized on first use
 }
+
+// SpanDetail summarizes the identifying fields of a Span, for
+// diagnostic and implementation-specific purposes.
+type SpanDetail struct {
```

Comment:
@yurishkuro Looking back at my previous comment, I just realized that I forgot that `spanImpl` is unexported, so you can't cast it outside of the package anyways. 

I'm +1 on the interface approach, since it'd be nice to avoid adding extra exported structs to the package.